packages/taco/src/conditions/compound-condition.ts

```diff
@@ -6,23 +6,45 @@ import { timeConditionSchema } from './base/time';
 
 export const CompoundConditionType = 'compound';
 
-export const compoundConditionSchema: z.ZodSchema = z.object({
-  conditionType: z
-    .literal(CompoundConditionType)
-    .default(CompoundConditionType),
-  operator: z.enum(['and', 'or']),
-  operands: z
-    .array(
-      z.lazy(() =>
-        z.union([
-          rpcConditionSchema,
-          timeConditionSchema,
-          contractConditionSchema,
-          compoundConditionSchema,
-        ]),
-      ),
-    )
-    .min(2),
-});
+export const compoundConditionSchema: z.ZodSchema = z
+  .object({
+    conditionType: z
+      .literal(CompoundConditionType)
+      .default(CompoundConditionType),
+    operator: z.enum(['and', 'or', 'not']),
+    operands: z
+      .array(
+        z.lazy(() =>
+          z.union([
+            rpcConditionSchema,
+            timeConditionSchema,
+            contractConditionSchema,
+            compoundConditionSchema,
+          ]),
+        ),
+      )
+      .min(1),
+  })
+  .refine(
+    (condition) => {
+      // 'and' and 'or' operators must have at least 2 operands
+      if (['and', 'or'].includes(condition.operator)) {
+        return condition.operands.length >= 2;
+      }
+
+      // 'not' operator must have exactly 1 operand
+      if (condition.operator === 'not') {
+        return condition.operands.length === 1;
```

Comment:
Yes. `.not` is a helper method and it exposes a proper interface, but there are ways around that - Including but not limited to calling the constructor directly.

Reply:
👍 . Had to resolve the same issue in `python` (i.e. there is also `not` helper class that has the proper interface for a single operand, but the base CompoundCondition class needed to validate operands length in case the object was created directly without the helper) hence the curiosity.